testing/test_collection.py

```diff
@@ -374,6 +374,26 @@ def pytest_configure(config):
         assert result.ret == 0
         assert "passed" in result.stdout.str()
 
+    def test_collectignoreglob_exclude_on_option(self, testdir):
+        testdir.makeconftest(
+            """
+            collect_ignore_glob = ['*w*l[dt]*']
+            def pytest_addoption(parser):
+                parser.addoption("--XX", action="store_true", default=False)
+            def pytest_configure(config):
+                if config.getvalue("XX"):
+                    collect_ignore_glob[:] = []
+        """
+        )
+        testdir.makepyfile(test_world="def test_hello(): pass")
+        testdir.makepyfile(test_welt="def test_hallo(): pass")
+        result = testdir.runpytest()
+        assert result.ret == EXIT_NOTESTSCOLLECTED
+        assert "passed" not in result.stdout.str()
```

Comment:
Probably better to explicitly use `result.stdout.fnmatch_lines('* no tests collected *')` as this is more explicit